testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToHudi.java

```diff
@@ -0,0 +1,264 @@
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): line 1:1: Table " + hiveTable.fullName() + " is redirected to " + hudiTable.fullName() + " and SET TABLE AUTHORIZATION is not supported with table redirections");
+
+        onHudi().executeQuery("DROP TABLE " + hudiTable.schemaTableName());
+    }
+
+    @Test(groups = {HIVE_HUDI_REDIRECTIONS, PROFILE_SPECIFIC_TESTS})
+    public void testDeny()
+    {
+        String tableName = "hudi_deny_" + randomNameSuffix();
+        TestTableName hudiTable = new TestTableName("hudi", "default", tableName);
+        TestTableName hiveTable = new TestTableName("hive", "default", tableName);
+
+        createHudiTable(hudiTable.schemaTableName(), HUDI_TABLE_TYPE_COPY_ON_WRITE, false);
+
+        assertQueryFailure(() -> onTrino().executeQuery("DENY DELETE ON " + hiveTable.fullName() + " TO ROLE PUBLIC"))
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): line 1:1: Table " + hiveTable.fullName() + " is redirected to " + hudiTable.fullName() + " and DENY is not supported with table redirections");
+
+        onHudi().executeQuery("DROP TABLE " + hudiTable.schemaTableName());
+    }
+
+    private static void assertResultsEqual(QueryResult first, QueryResult second)
+    {
+        assertThat(first).containsOnly(second.rows().stream()
+                .map(QueryAssert.Row::new)
+                .collect(toImmutableList()));
+
+        // just for symmetry
+        assertThat(second).containsOnly(first.rows().stream()
+                .map(QueryAssert.Row::new)
+                .collect(toImmutableList()));
+    }
+
+    private void createHudiTable(String tableName, String tableType, Boolean partitioned)
+    {
+        if (partitioned) {
+            createHudiPartitionedTable(tableName, bucketName, tableType);
+            return;
+        }
+
+        createHudiNonPartitionedTable(tableName, bucketName, tableType);
+    }
+
+    private static void createHudiNonPartitionedTable(String tableName, String bucketName, String tableType)
+    {
+        onHudi().executeQuery(format(
+                """
+                        CREATE TABLE %s (
+                          id bigint,
+                          name string,
+                          price int,
+                          ts bigint)
+                        USING hudi
+                        TBLPROPERTIES (
+                          type = '%s',
+                          primaryKey = 'id',
+                          preCombineField = 'ts')
+                        LOCATION 's3://%s/%s'""",
+                tableName,
+                tableType,
+                bucketName,
+                tableName));
+
+        onHudi().executeQuery("INSERT INTO " + tableName + " VALUES (1, 'a1', 20, 1000), (2, 'a2', 40, 2000)");
+    }
+
+    private static void createHudiPartitionedTable(String tableName, String bucketName, String tableType)
```

Comment:
`tableType` is always `HUDI_TABLE_TYPE_COPY_ON_WRITE` - we should just inline it within the method.

Reply:
both are inlined now.